internal/metrics/metrics.go

```diff
@@ -90,80 +88,51 @@ func (e *emitter) store() {
 			return
 		}
 
-		identifier := data.Key[0]
 		key := strings.Join(data.Key, ".")
 
-		_, keyExists := e.metricsData[identifier]
-		if !keyExists {
-			e.metricsData[identifier] = make(map[string]float64)
-		}
-
 		//register the metric with go-metrics,
 		//the metric key is used as the identifier.
-		_, registryExists := goMetricsRegistry[identifier]
+		_, registryExists := goMetricsRegistry[key]
```

Comment:
Because of this (checking `goMetricsRegistry[identifier]` but writing to `goMetricsRegistry[key]`) this _always_ was set to `false` and switch below run every time.